### ISSUE TITLE
JsonPath matching does not work anymore

### ISSUE THREAD

**Issue by BigHoss** [getinsomnia/insomnia#4237]
### Expected Behavior

jsonPath ```$..[?(@.Name.match('Doe'))]``` the result should only show the 2 doe entries

```json
[
  {
    "Id": "c6cb074b-e4eb-8534-d659-be2d2c1a29a0",
    "Name": "John Doe 123",
    "Serial": 1069777360,
    "DeviceEnergyType": 1,
    "MeterSubType": 3,
    "FamilyType": 1001,
    "ActivePower": 0.0,
    "ActivePowerUnit": "kW",
    "CounterReading": 1545.62,
    "CounterReadingUnit": "kWh",
    "CounterReadingT1": 1545.62,
    "CounterReadingImport": 1545.62,
    "ValueDate": "2021-11-22T15:51:33.8083885Z"
  },
  {
    "Id": "a9260872-64d1-22af-2a78-9451021ca4dc",
    "Name": "Jane Doe 123",
    "Serial": 22610343,
    "DeviceEnergyType": 1,
    "MeterSubType": 3,
    "FamilyType": 1001,
    "ActivePower": 0.0,
    "ActivePowerUnit": "kW",
    "CounterReading": 88.905,
    "CounterReadingUnit": "kWh",
    "CounterReadingT1": 88.905,
    "CounterReadingImport": 88.905,
    "ValueDate": "2021-11-22T15:51:29.9646345Z"
  },
  {
    "Id": "9dfa944e-adbd-0ad5-b391-11ffb4eb80e8",
    "Name": "Karl User 123",
    "Serial": 21783681,
    "DeviceEnergyType": 1,
    "MeterSubType": 3,
    "FamilyType": 1001,
    "ActivePower": 0.0,
    "ActivePowerUnit": "kW",
    "CounterReading": 0.0,
    "CounterReadingUnit": "kWh",
    "CounterReadingT1": 0.0,
    "CounterReadingImport": 0.0,
    "ValueDate": "2021-11-22T15:51:30.5740389Z"
  }
]
```

### Actual Behavior

the preview shows ```[]```instead

### Reproduction Steps

1. Access api that returns unnamed array of objects
2. try to filter the results with match

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2021.6.0

### What operating system are you using?

Windows

### Operating System Version

21H1 (Build 19043.1288)

### Installation method

chocolatey

### Last Known Working Insomnia version

about 2 month ago

**Comment by guyhom** [comment]
I also have this bug. 
JsonPath does not return expected results when a filter of this type is set: `[?(@...`. 
Other types of jsonpath queries seems to work.


Version: Insomnia 2021.7.2
Release date: 12/9/2021
OS: Darwin x64 19.6.0
Electron: 11.2.3
Node: 12.18.3
V8: 8.7.220.31-electron.0
Architecture: x64
node-libcurl: libcurl/7.73.0 OpenSSL/1.1.1k zlib/1.2.11 brotli/1.0.9 zstd/1.4.9 libidn2/2.1.1 libssh2/1.9.0 nghttp2/1.42.0

**Comment by dimitropoulos** [comment]
sorry about this!  I wish there was more we can do, but we're sorta beholden to the state of JSONPath, I'm sad to say.

please see https://github.com/Kong/insomnia/issues/4213#issuecomment-967353033

TLDR, use `` ` `` to escape the `@`

**Comment by BigHoss** [comment]
@dimitropoulos might be switching to [jsmespath](https://github.com/jdevillard/JmesPath.Net) be a solution
